ansible_collections/arista/avd/roles/eos_designs/python_modules/connected_endpoints/utils.py

```diff
@@ -104,8 +104,11 @@ def _get_short_esi(self, adapter: dict, channel_group_id: int, short_esi: str =
         # short_esi is only set when called from sub-interface port-channels.
         if short_esi is None:
             # Setting short_esi under port_channel will be removed in AVD5.0
-            port_channel_short_esi = get(adapter, "port_channel.short_esi")
-            if (short_esi := get(adapter, "ethernet_segment.short_esi", default=port_channel_short_esi)) is None:
+            if self._hostvars["type"] != "l2leaf":
```

Comment:
This is not how we should check for it. Types are customizable, so we have to check for capabilities instead. I suggest checking this:
```suggestion
            if self.shared_utils.overlay_evpn_vxlan and self.shared_utils.overlay_vtep:
```

Reply:
Also this will not capture all cases, since you add it below the short_esi is none check. Please move these two conditions (negated) to the `if` in line 100. So we return None if these two are NOT true.

Reply:
My bad. It should also be for evpn-mpls so I will update my comment above

Reply:
if self.shared_utils.overlay_evpn and self.shared_utils.overlay_vtep:
 
 can we write 'or' instead of 'and'?

Reply:
you can implement it any way you like. Just make sure you understand the logic and how it would work first.

Reply:
Fixed.